### PR TITLE
refactor: using error is instead of == [skip changelog]

### PR DESCRIPTION
**Pull request by kehiy**
I used `errors.Is()` instead of `==` or `!=` and also use the `%w` for wrapped errors.

### CODE REVIEW

**Review by Jorropo — CHANGES_REQUESTED** [review]
I wont comment on all the occurences, but `io.EOF` must never be wrapped because it was created before `errors.Is` and there is way to much code that relies on `==`.
`errors.Is` also generates garbage so we shouldn't use it unless it is realistical to trigger where `==` would not (so not `io.EOF`).

File: client/rpc/apifile.go [hunk 1]
```diff
@@ -94,7 +95,7 @@ func (f *apiFile) ReadAt(p []byte, off int64) (int, error) {
 	defer resp.Output.Close()
 
 	n, err := io.ReadFull(resp.Output, p)
-	if err == io.ErrUnexpectedEOF {
+	if errors.Is(err, io.ErrUnexpectedEOF) {
```
Comment (Jorropo) [inline client/rpc/apifile.go:98]: This also does nothing and generate garbage.

**Comment by kehiy** [comment]
@Jorropo so, I need to remove all `errors.Is()` ? or just for `io.EOF`s?

**Comment by Jorropo** [comment]
`io.EOF` and probably `io.ErrUnexpectedEOF` too

**Comment by kehiy** [comment]
@Jorropo its fixed

**Comment by kehiy** [comment]
@Jorropo can you have a review?

**Review by Jorropo — CHANGES_REQUESTED** [review]
Last one please and lgtm

**Comment by kehiy** [comment]
@Jorropo donee

**Review by Jorropo — APPROVED** [review]
LGTM thx

**Pull request merged**